inabox/tests/setup_infra.go

```diff
@@ -0,0 +1,211 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/inabox/deploy"
+	"github.com/Layr-Labs/eigenda/test/testbed"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/testcontainers/testcontainers-go/network"
+)
+
+// SetupGlobalInfrastructure creates the shared infrastructure that persists across all tests.
+// This includes containers for Anvil, LocalStack, GraphNode, and the Churner server.
+// This should be called once in TestMain.
+func SetupGlobalInfrastructure(
+	templateName, testName string,
+	inMemoryBlobStore bool,
+	logger logging.Logger,
+) (*InfrastructureHarness, error) {
+	infra := &InfrastructureHarness{
+		TemplateName:        templateName,
+		TestName:            testName,
+		InMemoryBlobStore:   inMemoryBlobStore,
+		Logger:              logger,
+		MetadataTableName:   "test-BlobMetadata",
+		BucketTableName:     "test-BucketStore",
+		MetadataTableNameV2: "test-BlobMetadata-v2",
+	}
```

Comment:
these things are input to the building of a infraHarness. take them out of the infraHarness and into a infraHarnessConfig or something. Then setupLocalInfrastructure below will take an infraHarnessConfig and output an infraHarness. Way cleaner than passing in a half-built infraHarness that gets modified in-place.